commands/bake.go

```diff
@@ -29,7 +28,11 @@ type bakeOptions struct {
 	printOnly  bool
 	sbom       string
 	provenance string
-	controllerapi.CommonOptions
```

Comment:
Would it be possible to split the changes to the common options out into a separate PR so we can merge those in asap? Those changes look good to me, and then we can unblock some other stuff.

I'm not 100% sure about the progress changes here, the progress config doesn't seem like it'll catch everything on the remote side.

I *think* what we want is a way of exposing a `progress.Writer` over the network, so that the client can pass a `progress.Writer` through `controller.Build`. Then we don't also have to pass a statusChan, since we can use `progress.Tee` if we need the individual solve status events. Then everything that gets written to that progress.Writer, either on the client, the local controller, or the remote controller will end up printed (even if it's contents written directly to the progress writer by e.g. the driver loader).

Reply:
Essentially, something like: https://github.com/docker/buildx/compare/master...jedevc:buildx:ktock-progress-printer (the remote side isn't properly implemented, and just creates a dummy printer, and then `Tee`s out the channel to transmit over the connection).

We should create the progress printer outside of each controller call, and pass it into the interface. Then we can lift `printWarnings` out of the controller entirely :tada: (should make things a bit neater).

Only thing I'm not sure about is how to handle the description: https://github.com/jedevc/buildx/blob/ktock-progress-printer/controller/build/build.go#L185-L193 - we need to have this info ready when we create the progress writer, so we need to load the drivers client before calling `RunBuild` (so maybe this is blocked on doing that?).

Reply:
We could also supplement `progress.Printer` with a new `Println` method or similar, that just prints out the content after progress has completed - we can then use this to handle the `PrintFunc` case.